Quaternion/tests/test_all.py

```diff
@@ -309,6 +309,35 @@ def test_dq_vectorized():
         q1.dq(q2.transform)
 
 
+def test_vector_to_scalar_correspondence():
+    """
+    Simple test that all possible transform pathways give the same
+    answer when done in vectorized form as they do for the scalar version.
+    """
+    atol = 1e-12
+
+    # Input equatorial has roll not in 0:360, so fix that for comparisons.
+    eq_23 = equatorial_23.copy()
+    normalize_angles(eq_23[..., -1], 0, 360)
+
+    # Compare vectorized computations for all possible input/output combos
+    # with the same for the scalar calculation.
+    q = Quat(equatorial=equatorial_23)
+    assert np.all(q.q == q_23)
+    assert np.all(q.equatorial == equatorial_23)
+    assert np.all(q.transform == transform_23)
+
+    q = Quat(q=q_23)
+    assert np.all(q.q == q_23)
+    assert np.allclose(q.equatorial, eq_23, rtol=0, atol=atol)
+    assert np.allclose(q.transform, transform_23, rtol=0, atol=atol)
+
+    q = Quat(transform=transform_23)
+    assert np.allclose(q.q, q_23, rtol=0, atol=atol)
+    assert np.allclose(q.equatorial, eq_23, rtol=0, atol=atol)
+    assert np.all(q.transform == transform_23)
+
+
```

Comment:
all these test were already there:
- The first three assert are in [line 131 and on](https://github.com/sot/Quaternion/blob/c2c7bfc690407cbfac9016b8344a0c8fa348d026/Quaternion/tests/test_all.py#L131)
- the next three (including the angle normalization) are in [line 224 and on](https://github.com/sot/Quaternion/blob/c2c7bfc690407cbfac9016b8344a0c8fa348d026/Quaternion/tests/test_all.py#L224)
- the last three are in [line 195 and on](https://github.com/sot/Quaternion/blob/c2c7bfc690407cbfac9016b8344a0c8fa348d026/Quaternion/tests/test_all.py#L195)

Having them together means the angle normalization is done only once.

Reply:
Ah I see, excellent.  But it doesn't hurt to have redundant tests, and these are organized in a way that works for my brain.  As long as you agree these tests are valid then I'd like to keep them in.

Reply:
I'm ok with that.